### DISCUSSION TITLE
Feature Requests

### DISCUSSION THREAD

**Discussion by lavron** [lavron/docker-voltronic-mqtt#2] [Ideas]
These are currently supported features and I'm interested in your thoughts on new ones to add. What do you think could make this project more useful for you? 

Please share your suggestions below, being clear and concise with your feature descriptions.

Thank you in advance!

Best,
Viktor

**Comment by 1967luca** [comment]
Buongiorno ho usato il tuo progetto con home assistant core + mosquitto ma nonostante legga i dati dall'inverter sul terminale facendo docker logs come mai i miei sensori mostrano sempre sconosciuto? scusa la domanda può essere stupida ma il file config penso di averlo fatto bene grazie

**Reply by lavron** [reply]
Hey!

You see the data in the terminal - that means the inverter is connected to mqtt and sending data. Something is wrong between mqtt and Home Assistant. 

Please check in HA the mqtt messages you receiving: subscribe to `homeassistant/sensor/#` ([how to do that](https://community.home-assistant.io/t/how-to-listen-to-mqtt-topic-under-dev-tools/422762/2)) and check the output.
You should see something like 
```
Message 279 received on homeassistant/sensor/inverter_pv_input_watt at 7:59:
12 
```
If you don't see it - HA might be disconnected from mqtt. 
If you see it, but in HA sensors still values "unknown" - it might be something wrong with "configuration.yaml". Check mqtt topics.

Let me know how it worked for you.

**Comment by rmuk** [comment]
Hi there. Thanks for all your work on this. I've got a Voltacon-branded device that's connected to my Raspberry Pi via USB and it presents as /dev/hidraw0 rather than /dev/ttyUSBx. When I try to install I get an error and I'm assuming it's because I'm trying to open a HID device as a serial device. Is this something you can easily resolve? Unfortunately the setup is very remote and I can't get there easily, otherwise I'd find a USB-to-Serial adapter and switch to using that (and is it too much to hope that the interver itself would emulate a serial device rather than HID? Anyway...).

```
sudo docker run -t -i --privileged -v /dev:/dev --restart=always --name voltronic-mqtt --pull=always -e MQTT_PASSWORD='xxxxxx' -e MQTT_SERVER='localhost:1883' -e MQTT_USER='xxxxxx' -e SERIAL_PORT='/dev/hidraw0' lavron/voltronic-mqtt:latest
latest: Pulling from lavron/voltronic-mqtt
Digest: sha256:34012bda4f58e9f49aa8967b3ca0970a201b5de88a4b3cc3d0400ec36ddebde4
Status: Image is up to date for lavron/voltronic-mqtt:latest
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/serial/serialposix.py", line 398, in _reconfigure_port
    orig_attr = termios.tcgetattr(self.fd)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
termios.error: (22, 'Invalid argument')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/./main.py", line 9, in <module>
    inverter = Voltronic()
               ^^^^^^^^^^^
  File "/app/voltronic.py", line 13, in __init__
    self.port = SerialPort()
                ^^^^^^^^^^^^
  File "/app/serial_port.py", line 9, in __init__
    self.port = serial.Serial(SERIAL_PORT, 2400, timeout=0.5)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/serial/serialutil.py", line 244, in __init__
    self.open()
  File "/usr/local/lib/python3.11/site-packages/serial/serialposix.py", line 332, in open
    self._reconfigure_port(force_update=True)
  File "/usr/local/lib/python3.11/site-packages/serial/serialposix.py", line 401, in _reconfigure_port
    raise SerialException("Could not configure port: {}".format(msg))
serial.serialutil.SerialException: Could not configure port: (22, 'Invalid argument')
```

**Reply by kryspirit** [reply]
you doesn't has de usb device connected and recognized so python serialPort initialize doesn't work
